umodbus/async/tcp.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Pycom Limited.
+#
+# This software is licensed under the GNU GPL version 3 or any
+# later version, with permitted additional terms. For more information
+# see the Pycom Licence v1.0 document supplied with this file, or
+# available at https://www.pycom.io/opensource/licensing
+#
+
+# system packages
+try:
+    import uasyncio as asyncio
+except ImportError:
+    import asyncio
+
+from ..sys_imports import struct, List, Optional, Callable, Coroutine, Any
+
+# custom packages
+from .modbus import AsyncModbus
+from .common import AsyncRequest, CommonAsyncModbusFunctions
+from .. import functions, const as Const
+from ..common import ModbusException
+from ..tcp import CommonTCPFunctions, TCPServer
+
+
+class AsyncModbusTCP(AsyncModbus):
+    def __init__(self, addr_list: Optional[List[int]] = None):
+        super().__init__(
+            # set itf to AsyncTCPServer object
+            AsyncTCPServer(),
+            addr_list
+        )
+
+    async def bind(self,
```

Comment:
doc strings are missing for this file and there might be some flake8 warnings ;)